### PR TITLE
Add more detail to hoppity pv page

### PR DESCRIPTION
**Pull request by catgirlseraid**
dependant on
- #1157 

added tooltips and a bunch of other changes to the above pr

### CODE REVIEW

**Review by NopoTheGamer — APPROVED** [review]

File: src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/profileviewer/HoppityPage.kt [hunk 1]
```diff
@@ -800,6 +850,53 @@ class HoppityPage(pvInstance: GuiProfileViewer) : GuiProfileViewerPage(pvInstanc
...
+                            "§6Prestige Chocolate Realm\n" +
+                                "§7Grants §a+5% §7chance to find a\n" +
+                                "§7§aChocolate Rabbit §7that you haven't\n" +
+                                "§7found yet and grants §6+50 Chocolate\n" +
+                                "§6§7per second.\n" +
+                                "\n" +
+                                "§6§lLEGENDARY ACCESSORY"
+                        }
+
+                        else -> ""
```
Comment (nea89o) [inline src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/profileviewer/HoppityPage.kt:897]: This looks like a great candidate for using repo data. Perhaps you could grab data from the existing talismen?

**Review by nea89o — DISMISSED** [review]

File: src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/profileviewer/HoppityPage.kt (hunk 1 above)
Comment (nea89o) [reply src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/profileviewer/HoppityPage.kt:897]: You can even grab the tooltip of `shownTalismanItem` which would even include information about recombs and such, i think.

Comment (catgirlseraid) [reply src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/profileviewer/HoppityPage.kt]: resolved in 3424cbf

File: src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/profileviewer/HoppityPage.kt [hunk 2]
```diff
@@ -244,13 +308,133 @@ class HoppityPage(pvInstance: GuiProfileViewer) : GuiProfileViewerPage(pvInstanc
...
+
+                    UpgradeType.TIME_TOWER -> {
+                        if (upgradeInfo.level == 0) {
+                            fallbackList(upgradeInfo.displayName)
+                        } else {
+                            buildList {
+                                add("§d${upgradeInfo.displayName} ${upgradeInfo.level.toRoman()}")
+                                add("")
+                                add("§6+${((upgradeInfo.level * 0.1) + muTimeTowerBonus).roundToDecimals(1)}x Chocolate §7per second for §a1h§7.")
+                                add("§7Time Tower charges §a$timeTowerCharges§7/§a3")
```
Comment (jani270) [inline src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/profileviewer/HoppityPage.kt:346]:
```suggestion
                                add("§7Time Tower charges: §a$timeTowerCharges§7/§a3")
```

**Review by NopoTheGamer — APPROVED** [review]

**Review by ThatGravyBoat — APPROVED** [review]

File: src/main/kotlin/io/github/moulberry/notenoughupdates/util/KotlinNumberUtils.kt [hunk 3]
```diff
@@ -19,10 +19,37 @@
...
+        10 to "X",
+        9 to "IX",
+        5 to "V",
+        4 to "IV",
+        1 to "I",
+    )
+)
+
+fun Int.toRoman(): String {
+    if (this <= 0) error("$this must be positive!")
```
Comment (Lulonaut) [inline src/main/kotlin/io/github/moulberry/notenoughupdates/util/KotlinNumberUtils.kt:50]: While it doesn't look to be possible with the current usage, I don't think crashing the game is preferable over simply returning an error string

**Review by Lulonaut — APPROVED** [review]

**Pull request merged**